### ISSUE TITLE
System.IndexOutOfRangeException v0.11.3

### ISSUE THREAD

**Issue by OrbintSoft** [dotnet/BenchmarkDotNet#974]
I get this exception when I try to run my benchmarks:

```
		$exception	{System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at BenchmarkDotNet.Reports.Measurement.Parse(ILogger logger, String line, Int32 processIndex, Encoding encoding)}	System.IndexOutOfRangeException
```

I am a bit new to this library, so maybe I am doing something wrong, but the code is very simple.

        public static void Main(string[] args)
        {
            BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);
        }

All tests runs without errors, until in console is printed "// AfterAll"

**Comment by OrbintSoft** [comment]
I tried to downgrade and it seems the bug has appeared starting from version 0.11.2, with version 0.11.1 all works fine

**Comment by wojtpl2** [comment]
@OrbintSoft Can you share the source code of your benchmark?

@adamsitnik @AndreyAkinshin In my short story with BenchmarkDotNet it is second issue where BDN got exception during calculating statistics. Maybe will be good to catch this issue and show "Error" in column and print error log with comment: "An exception occured during calculating X. Please create new issue on Github"

**Comment by OrbintSoft** [comment]
@wojtpl2 sure the project is open source: https://github.com/OrbintSoft/yauaa.netstandard/tree/master/OrbintSoft.Yauaa.NetStandard/OrbintSoft.Yauaa.NetCore.Benchmarks

But I doubt the problem is in my code, the issue has been fixed just downgrading to version 0.11.1

If you can catch the issue maybe is better to add some unit testing instead to log the error and  demand to the user to report it.

**Comment by adamsitnik** [comment]
> If you can catch the issue maybe is better to add some unit testing instead to log the error and demand to the user to report it.

we always add a unit test and fix the bug, what @wojtpl2 was to also add a catch block to make sure the benchmark does not fail for a test case that we don't know yet but could somehow make it fail

@wojtpl2 personally I prefer a fail fast approach. The best thing here would be to just fix it, I am not really into adding defensive catch blocks

**Comment by wojtpl2** [comment]
@adamsitnik You have right. Usage of BDN is different. And it encourages users to report bugs.

**Comment by OrbintSoft** [comment]
Sure, however I just realized that maybe is a false problem, the exception is catched since 0.11.3.
I encountered the exception for first time with 0.11.2,  that was the first time that I used BenchmarkDotNet, thinking that was a problem in my code I enabled debugger.
In 0.11.2 the exception is not catched and the program fails so bencharking stop execution.
Maybe is related to this  issue that is fixed: #https://github.com/dotnet/BenchmarkDotNet/issues/950 
After upgrade to version 0.11.3, I saw the same exception in debugger, because I attached the debugger to see where it fails.
Due to release mode breakpoint was misleading at end of main, and I thought the program was failing.
But I haven't realized that with the new version the program don't fail to execute, just click continue some times  and the program will continue the execution completing all benchmarks.
The exception is thrown one time for each benchmark, starting from "After All" phase.
Sorry if I made you lose time.

Here is the log if still useful:
[OrbintSoft.Yauaa.Benchmarking.SplitterBenchmarks.log](https://github.com/dotnet/BenchmarkDotNet/files/2626243/OrbintSoft.Yauaa.Benchmarking.SplitterBenchmarks.log)
